hashicorp/waypoint
# Backport of Add sorting to `serverstate.PipelineRunList` into release/0.11.x

## Backport

This PR is auto-generated from #4797 to be assessed for backporting due to the inclusion of the label backport/0.11.x.


:rotating_light:
>**Warning** automatic cherry-pick of commits failed. If the first commit failed,
you will see a blank no-op commit below. If at least one commit succeeded, you
will see the cherry-picked commits up to, _not including_, the commit where
the merge conflict occurred.

The person who merged in the original PR is:
@jgwhite
This person should manually cherry-pick the original PR into a new backport PR,
and close this one when the manual backport PR is merged in.

> merge conflict error: POST https://api.github.com/repos/hashicorp/waypoint/merges: 409 Merge conflict []



The below text is copied from the body of the original PR.

---

## Why the change?

We’d like to be able to show runs newest-first in the UI.

Note that this PR only addresses sorting at the state layer. Longer term, we hope to [allow sorting options](https://github.com/hashicorp/waypoint/blob/WAYP-1541/pkg/server/proto/server.proto#L581) to be passed to `UI_ListPipelineRuns`, but that’s for another PR.

## What does it look like?

```sh
$ ./contrib/waypoint-grpc/waypoint-grpc.sh ListPipelineRuns '{ "pipeline": { "id": "01H2QGNVQ77KB4H67VJDESEATA" } }'
{
  "pipelineRuns": [
    {
      "sequence": "1",
      // ...
    },
    {
      "sequence": "2",
      // ...
    },
    {
      "sequence": "3",
      // ...
    }
  ]
}

$ ./contrib/waypoint-grpc/waypoint-grpc.sh UI_ListPipelineRuns '{ "pipeline": { "id": "01H2QGNVQ77KB4H67VJDESEATA" } }'
{
  "pipelineRunBundles": [
    {
      "pipelineRun": {
        "sequence": "3",
        // ... 
      },
    },
    {
      "pipelineRun": {
        "sequence": "2",
        // ...
      },
    },
    {
      "pipelineRun": {
        "sequence": "1",
        // ...
      },
    }
  ],
}
```

## How do I test it?

1. Boot the server
2. Run a pipeline a few times
3. List runs using `ListPipelineRuns`
4. Verify they are in insertion-order
5. List runs using `UI_ListPipelineRuns`
6. Verify they are in newest-first order

---

<details>
<summary> Overview of commits </summary>

  - d1d41ca62fedd5990297edbc1f1dc04640791b94  - 70856069d42eb43d469021415b3f66004a227999 

</details>
